src/overlays/actors/ovl_Boss_05/z_boss_05.c

```diff
@@ -159,107 +341,1352 @@ ActorInit Boss_05_InitVars = {
+            this->knockbackMagnitude = -12.0f;
+            this->knockbackAngle = this->dyna.actor.yawTowardsPlayer;
+            this->damagedTimer = 6;
+            return;
+        }
+
+        this->damagedTimer = 10;
+        this->dyna.actor.speed = 0.0f;
+        this->knockbackMagnitude = -20.0f;
+        this->knockbackAngle = this->dyna.actor.yawTowardsPlayer;
+
+        Actor_PlaySfx(&this->dyna.actor, NA_SE_EN_MIZUBABA2_DAMAGE);
+
+        switch (this->dyna.actor.colChkInfo.damageEffect) {
+            case BIO_BABA_DMGEFF_STUN:
+                Boss05_WalkingHead_SetupStunned(this, play);
+                break;
+
+            case BIO_BABA_DMGEFF_FIRE:
+                this->drawDmgEffState = BIO_BABA_DRAW_DMGEFF_STATE_FIRE_INIT;
+                attackDealsDamage = true;
+                break;
+
+            case BIO_BABA_DMGEFF_FREEZE:
+                Boss05_WalkingHead_SetupFreeze(this, play);
+                this->drawDmgEffState = BIO_BABA_DRAW_DMGEFF_STATE_FROZEN_INIT;
+                break;
+
+            case BIO_BABA_DMGEFF_LIGHT_ORB:
+                this->drawDmgEffState = BIO_BABA_DRAW_DMGEFF_STATE_LIGHT_ORB_INIT;
+                attackDealsDamage = true;
+                break;
+
+            default:
+                attackDealsDamage = true;
+                break;
+        }
+
+        if (attackDealsDamage) {
+            u8 damage;
+
+            if ((this->actionFunc == Boss05_WalkingHead_Stunned) &&
+                (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) && (this->drawDmgEffTimer != 0)) {
+                Boss05_WalkingHead_Thaw(this, play);
+                this->drawDmgEffState = BIO_BABA_DRAW_DMGEFF_STATE_NONE;
+            }
+
+            damage = this->dyna.actor.colChkInfo.damage;
+            this->dyna.actor.colChkInfo.health -= damage;
+            if ((s8)this->dyna.actor.colChkInfo.health <= 0) {
+                Enemy_StartFinishingBlow(play, &this->dyna.actor);
+            }
+
+            Boss05_WalkingHead_SetupDamaged(this, play);
+            this->damagedFlashTimer = 15;
+        }
+    }
+}
+
+/**
+ * Returns true if the Bio Deku Baba's model is rotated such that it is roughly facing the player, false otherwise.
+ */
+s32 Boss05_WalkingHead_IsLookingAtPlayer(Boss05* this, PlayState* play) {
+    s16 yawDiff = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.shape.rot.y;
+
+    if (ABS_ALT(yawDiff) < 0x3000) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/**
+ * Checks to see if the Bio Deku Baba is looking at the player, if it's within 200 units of the player, and if the
+ * player is not too far above or below the Bio Deku Baba. If all of these conditions are met, it will setup the Bio
+ * Deku Baba to play its spotted animation and start charging towards the player.
+ */
+void Boss05_WalkingHead_CheckIfPlayerWasSpotted(Boss05* this, PlayState* play) {
```

Comment:
I don't know if it reads better in this particular case, but when there's a function that will setup an action but only based on a condition, I've been using the work `try` for those cases. So here, it would be `Boss05_WalkingHead_TrySpottingPlayer`. What do you think of that?

Reply:
I think that makes sense, thanks for the suggestion. Let me change it.